### DISCUSSION TITLE
UBO ignoring a large % of filters in Adguard's base filter

### DISCUSSION THREAD

**Discussion by BillyJoeJimBob** [uBlockOrigin/uAssets#13152] [Q&A] [answered]
### Prerequisites

- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uAssets/issues) to avoid opening a duplicate issue
- [X] I am running the latest version of uBlock Origin

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

### URL(s) where the issue occurs

N/A, general filter issue.

### Describe the issue

UBO is ignoring a large number of filters in Adguard's base filter.  When I use filters which include the Adguard base filter (UBO's built-in AG base filter or imported AG base+Easylist filters), 10-25% of the filters are not being counted in UBO's total network + cosmetic filter list count.  This occurs with all other filters disabled, otherwise default UBO configuration.  Is this the expected behavior?  

### Screenshot(s)

<img width="467" alt="AG base" src="https://user-images.githubusercontent.com/11702508/167614600-21a12ca4-285c-4d99-aad7-76fd39f510d7.png">
<img width="448" alt="AG base + EZL" src="https://user-images.githubusercontent.com/11702508/167614631-d6da0eb1-185c-4220-9415-eff9b2edefb2.png">
<img width="545" alt="AG base + EZL optimized" src="https://user-images.githubusercontent.com/11702508/167614661-2c05ce82-e307-49f3-8ed8-22c655484a0f.png">

### uBlock Origin version

1.42.4

### Browser name and version

Firefox 100.0

### Settings

All other UBO filters disabled.

Enabled, one of the following AG filters: 

1) UBO's included AdGuard Base filter, or

2) EasyList + AdGuard English filter.  []https://filters.adtidy.org/extension/ublock/filters/2.txt

or

3) AdGuard Base filter + EasyList (Optimized).  []https://filters.adtidy.org/extension/ublock/filters/2_optimized.txt

### Notes

I searched in Github and Reddit for answers to this.  My understanding is UBO ignores filters that are not in proper syntax, however AG base filter's 21% seemed like a high rejection rate for an optional filter included with UBO.

**Comment by u-RraaLL** [comment]
Is this some counter bug @gorhill ?

![image](https://user-images.githubusercontent.com/92610420/167669575-6c9c9fd2-49f6-4636-b3f9-c062ef81e11d.png)

uBO filter list the only list selected as a test.

Out of the under 32k filters "used", only over 14k are counted above. Is the rest scriptlets, are these not counted?

Btw. Is there 29 duplicates within the list? Is that why 29 is not used?

**Reply by gorhill** [reply, marked as answer]
Scriptlet filters and HTML filters are not counted as either network or cosmetic filters.
